### PR TITLE
Segment integrations

### PR DESCRIPTION
**Pull request by alex-tpom6oh**

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝Changed routes:


| [**…ons/segment-javascript**](https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment-javascript) | [**…nations/segment-kotlin**](https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment-kotlin) |
|:-:|:-:|
|<a href="https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment-javascript">Screenshot of /implementation/destinations/segment-javascript</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/_4vmvVhCxITllZGUZVFaiNNHWdIoKtFCm48gm-r-NKn6TgByUBgz4rzo-c2SIYJxKJ4N480pKJ0OkjeWOmkCWg.png">(view full size)</a> |<a href="https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment-kotlin">Screenshot of /implementation/destinations/segment-kotlin</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/YX-z7ii-34Jld3Z5_ouMBoywHEAd8YHYz_AlZ-rB6iGO5XH-vatTILDxt55UU-7Q27txlMwmbt-zOZPkosVT5Q.png">(view full size)</a> |



| [**…inations/segment-swift**](https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment-swift) | [**…n/destinations/segment**](https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment) |
|:-:|:-:|
|<a href="https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment-swift">Screenshot of /implementation/destinations/segment-swift</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/G7yf5UHigo2gxf_kY1YC-J5Uy_ocWGAtXWG8Ap9qab5PgkyA-UF9b_bIIrSfA59YSiv65sHKnwWnGKcrn_rr5g.png">(view full size)</a> |<a href="https://docs-kkso5a6eb.avo-dev.app/implementation/destinations/segment">Screenshot of /implementation/destinations/segment</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/IIG4k8apaOrPGwcSNn-NVd3SiUhW5mtwAaD4Y4LCLj0V6OSM6SVeTfkS2C-_pSI2IVXFnSsD6zcUq7HqxITPbQ.png">(view full size)</a> |



| [**…egrations/integrations**](https://docs-kkso5a6eb.avo-dev.app/integrations/integrations) | [**/integrations/segment**](https://docs-kkso5a6eb.avo-dev.app/integrations/segment) |
|:-:|:-:|
|<a href="https://docs-kkso5a6eb.avo-dev.app/integrations/integrations">Screenshot of /integrations/integrations</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/93UxDOAY-2ubJRA-66dKNhSqleesVul7wiBC_9pxi-WRPeTQOZ3xcadbrc9OURm48jRuJoryGBcXqPXmDQH2FA.png">(view full size)</a> |<a href="https://docs-kkso5a6eb.avo-dev.app/integrations/segment">Screenshot of /integrations/segment</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/IwefL6n5nYMQNYTIX9KoN-xdkqW5pITTXQO0XYgMr-fArhw8d-CtrtCUsfkpoDxLQeWytcyf-tj9R8l-iq9ang.png">(view full size)</a> |







Commit 2cd23298d56e75eaab92e7023995e6fdcd31cd60 (https://docs-kkso5a6eb.avo-dev.app).

File: components/Navigation.tsx [hunk 1]
```diff
@@ -394,6 +389,40 @@ const navigation: navigationItem[] = [
...
+            title: 'Publish tracking plan to Protocols',
+            path: '/workspace/tracking-plan/publishing',
+          },
+          {
+            type: 'route',
+            title: 'Avo Functions to Segment',
+            path: '/implementation/destinations/segment',
+          },
+        ],
+      },
```
Comment (logason) [inline components/Navigation.tsx:423]: I think we should keep the integration specific pages as a one-pager overview of all integration points, with links to other places in the doc with more detail. For example the Inspector <> Segment would link to the "Tracking Observability" section and the "Reference > Inspector". That way we avoid duplicating content across multiple sections in the doc, and keep the Integration page focused on the overview of all the different integration options we have with each tool. Does that makes sense @alex-tpom6oh ?

File: pages/implementation/destinations/segment-javascript.mdx [hunk 2]
```diff
@@ -0,0 +1,138 @@
+---
+layout: 'guide'
+title: 'Use Segment in JavaScript'
+---
+
+Custom destination to send data from JavaScript to Segment
```
Comment (logason) [inline pages/implementation/destinations/segment-javascript.mdx:6]: Here we should also highlight that we do offer managed Segment SDK integration, same applies to the other Segment <> language docs

File: pages/implementation/destinations/segment.mdx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+layout: 'guide'
+title: 'Avo Functions to Segment'
+---
+
+import Link from '../../../components/Link';
+
+- <Link passHref href="/implementation/destinations/segment-javascript"><a>JavaScript</a></Link>
+- <Link passHref href="/implementation/destinations/segment-kotlin"><a>Kotlin</a></Link>
+- <Link passHref href="/implementation/destinations/segment-swift"><a>Swift</a></Link>
```
Comment (logason) [inline pages/implementation/destinations/segment.mdx:10]: This list indicates that Segment is not available in other languages, which is something we should avoid. While we don't have destination specific docs for each language, should we link to a more generic custom destination docs for all the other languages?

File: pages/integrations/integrations.mdx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+layout: 'guide'
+title: 'All integrations'
+---
+
+import Link from '../../components/Link';
+
+Here is a list of tools you can integrate Avo with
+
+<Link passHref href="/integrations/segment"><a>Segment</a></Link>
```
Comment (logason) [inline pages/integrations/integrations.mdx:10]: Here will be a perfect opportunity to highlight and explain the different integration types we offer (productivity, observability, implementation etc) and list out the tools that support each

File: pages/integrations/segment.mdx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## Send app data to Inspector with the SDK
+
+<Link passHref href="/implementation/avo-inspector-sdk-reference"><a>Stream app data to Inspector with an SDK</a></Link>
+
+## Publish tracking plan to Protocols
+
+<Link passHref href="/workspace/tracking-plan/publishing#segment-protocols"><a> Publish tracking plan to Protocols</a></Link>
+
+## Generate code to send data to Segment
```
Comment (logason) [inline pages/integrations/segment.mdx:20]: Here we should be highlighting the value props of each integration point

File: components/Navigation.tsx (hunk 1 above)
Comment (logason) [reply components/Navigation.tsx:423]: I'd like to suggest we hold off with adding this to the sidebar, until the integrations docs are more complete

**Review by logason — CHANGES_REQUESTED** [review]

File: pages/implementation/destinations/segment-javascript.mdx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    // called when you invoke Avo functions with the Unidentify User action
+    analytics.identify(null, {});
+  },
+  revenue: function(amount, eventProperties) {
+    // called when you invoke Avo functions with the Log Revenue action
+    analytics.track("revenue", {"revenue": amount});
+  },
+  page: function(pageName, eventProperties) {
+    // called when you invoke Avo functions with the Log Page View action
+    analytics.page(eventName, eventProperties);
```
Comment (alex-tpom6oh) [inline pages/implementation/destinations/segment-javascript.mdx:41]: should be `page(pageName...`

File: components/Navigation.tsx (hunk 1 above)
Comment (alex-tpom6oh) [reply components/Navigation.tsx:423]: sounds good

File: pages/integrations/segment.mdx (hunk 5 above)
Comment (alex-tpom6oh) [reply pages/integrations/segment.mdx:20]: Do you suggest doing it in this pr? Not sure what exactly to write here

Comment (logason) [reply pages/integrations/segment.mdx:20]: We should do it before we start multiplying this doc for other integration types. We can ship this now and iterate before we write integration docs for other destinations, or hold off to merging until we've nailed down the copy here

**Comment by hugihlynsson** [comment]
Is this PR still relevant? Should we wrap it up or close it?

**Comment by alex-tpom6oh** [comment]
Good question 😄

**Comment by hugihlynsson** [comment]
@alex-tpom6oh @logason is this PR still relevant? Should we put the finishing touches on it or close it?

**Pull request closed without merging**